### PR TITLE
[10.0][FIX] backport of SerpentCS mass_editing v10 port

### PR DESCRIPTION
**Pull request by jbchouinard**
I merged the v10 port of mass_editing by @JayVora-SerpentCS, found here: https://github.com/JayVora-SerpentCS/MassEditing/tree/10.0.

I split the code from the translations to make review easier.

Commit https://github.com/OCA/server-tools/commit/4614486f9b51ca779bc17b9bac9196a063dbebfc fixes issue https://github.com/OCA/server-tools/issues/806.

### CODE REVIEW

**Comment by jbchouinard** [comment]
@JayVora-SerpentCS I was looking at this while merging the code:

```
domain="[('ttype', 'not in', ['reference', 'function']), ('model_id', 'in', model_list)]"/>
```

What's the reason for not simply using ('model_id', '=', model_id)?

I dont really understand what model_list is for since it's just set to [model_id] in the onchange on model_id.

EDIT: Nevermind I hadn't looked carefully, it's right there in the code, the parent models are added to model_list, so sometimes it has more than one id.

File: mass_editing/i18n/da.po [hunk 1]
```diff
@@ -60,6 +66,9 @@ msgid ""
 msgstr ""
 
 #. module: mass_editing
+=======
+#: model:ir.model,name:mass_editing.model_ir_model_fields
+>>>>>>> c5fd7e3... Made Mass Editing module v10 compatible
```
Comment (sbidoul) [inline mass_editing/i18n/da.po]: @jbchouinard there is an unresolved conflict here.

**Comment by sbidoul** [comment]
Otherwise it works fine.

**Comment by ovnicraft** [comment]
is the end of world w/o this module !, is ready to merge ?

**Comment by JayVora-SerpentCS** [comment]
Should never ne model_list, rather. It was a backport from old version - https://github.com/OCA/server-tools/pull/598#issuecomment-273154564

**Comment by anhvu-sg** [comment]
Hello, can anyone merge this commit please ?

**Comment by RemiFr82** [comment]
Hello, I agree @anhvu2004 , can anyone merge this PR ?

**Comment by jbchouinard** [comment]
@sbidoul oops fixed that. I grepped and didnt find any more unresolved conflicts

**Comment by joaoalf** [comment]
:+1: LGTM

Please someone merge this as soon as possible.

**Review by max3903 — APPROVED** [review]

**Comment by JayVora-SerpentCS** [comment]
This shouldn't have been merged dears! Some edits revert the actual Pr #598 and the old cherry picked commits @pedrobaeza

**Comment by ovnicraft** [comment]
@JayVora-SerpentCS can you make a new PR ?

**Comment by JayVora-SerpentCS** [comment]
I guess better to uncommit this one and correct, otherwise we will go in another loop of complexity.

**Comment by sbidoul** [comment]
We cannot uncommit, only revert. Please do another PR.

**Comment by JayVora-SerpentCS** [comment]
I meant revert, sorry for the confusion.

**Pull request merged**